meta-mel/lib/bblayers/mel_utils.py

```diff
@@ -241,7 +242,22 @@ def do_dump_licenses(self, args):
                 pn = data.getVar('PN')
                 pv = data.getVar('PV')
                 lc = data.getVar('LICENSE')
-                f.write('%s,%s,%s\n' % (pn, pv, lc))
+
+                if not args.extrainfo:
+                    f.write('%s,%s,%s\n' % (pn, pv, lc))
+                else:
+                    # For SRC_URI we need to drop the local sources i.e. file:// entries
+                    # then for further filtering we pick up only the initial part of the archive/git
+                    # src i.e. dropping out bitbake specifics like protocol, branch, name etc.
+                    su = " ".join([x.split(';')[0] for x in data.getVar('SRC_URI').split() if not x.startswith('file://')])
```

Comment:
This isn't a good way to handle this. bb.fetch has methods for parsing the urls, determining the scheme, etc. We should be using that rather than direct string manipulation. You can directly split the list of urls and parse them with `bb.fetch.decodeurl`, or you can construct a fetch object, i.e. `bb.fetch.Fetch(data.getVar('SRC_URI').split(), data)` and access the url data on the fetch object.

Reply:
Done. Thanks!